src/domain/hooks/helpers/event-cache.helper.ts

```diff
@@ -0,0 +1,553 @@
+import { Inject, Injectable, Module } from '@nestjs/common';
+import {
+  BalancesRepositoryModule,
+  IBalancesRepository,
+} from '@/domain/balances/balances.repository.interface';
+import {
+  ChainsRepositoryModule,
+  IChainsRepository,
+} from '@/domain/chains/chains.repository.interface';
+import {
+  CollectiblesRepositoryModule,
+  ICollectiblesRepository,
+} from '@/domain/collectibles/collectibles.repository.interface';
+import {
+  IMessagesRepository,
+  MessagesRepositoryModule,
+} from '@/domain/messages/messages.repository.interface';
+import {
+  ISafeAppsRepository,
+  SafeAppsRepositoryModule,
+} from '@/domain/safe-apps/safe-apps.repository.interface';
+import {
+  ISafeRepository,
+  SafeRepositoryModule,
+} from '@/domain/safe/safe.repository.interface';
+import {
+  ITransactionsRepository,
+  TransactionsRepositoryModule,
+} from '@/domain/transactions/transactions.repository.interface';
+import {
+  TransactionEventType,
+  ConfigEventType,
+} from '@/routes/hooks/entities/event-type.entity';
+import { LoggingService, ILoggingService } from '@/logging/logging.interface';
+import { Event } from '@/routes/hooks/entities/event.entity';
+import {
+  BlockchainRepositoryModule,
+  IBlockchainRepository,
+} from '@/domain/blockchain/blockchain.repository.interface';
+import { IStakingRepository } from '@/domain/staking/staking.repository.interface';
+import { memoize, MemoizedFunction } from 'lodash';
+import { StakingRepositoryModule } from '@/domain/staking/staking.repository.module';
+
+@Injectable()
+export class EventCacheHelper {
+  private static readonly HOOK_TYPE = 'hook';
+  public isSupportedChainMemo: ((chainId: string) => Promise<boolean>) &
+    MemoizedFunction;
+
+  constructor(
+    @Inject(IBalancesRepository)
+    private readonly balancesRepository: IBalancesRepository,
+    @Inject(IBlockchainRepository)
+    private readonly blockchainRepository: IBlockchainRepository,
+    @Inject(IChainsRepository)
+    private readonly chainsRepository: IChainsRepository,
+    @Inject(ICollectiblesRepository)
+    private readonly collectiblesRepository: ICollectiblesRepository,
+    @Inject(IMessagesRepository)
+    private readonly messagesRepository: IMessagesRepository,
+    @Inject(ISafeAppsRepository)
+    private readonly safeAppsRepository: ISafeAppsRepository,
+    @Inject(ISafeRepository)
+    private readonly safeRepository: ISafeRepository,
+    @Inject(IStakingRepository)
+    private readonly stakingRepository: IStakingRepository,
+    @Inject(ITransactionsRepository)
+    private readonly transactionsRepository: ITransactionsRepository,
+    @Inject(LoggingService)
+    private readonly loggingService: ILoggingService,
+  ) {
+    this.isSupportedChainMemo = memoize(
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      this.chainsRepository.isSupportedChain.bind(this.chainsRepository),
+    );
+  }
+
+  public async onEventClearCache(event: Event): Promise<void[]> {
```

Comment:
I'm having a hard time understanding this method. What do you think about mapping the event type to the related method instead on having different `case`s? As an example:
```suggestion
  export const EventTypeHandler = {
    TransactionEventType.PENDING_MULTISIG_TRANSACTION: onTransactionEventPendingMultisigTransaction,
    // ...
  }
  public async onEventClearCache(event: Event): Promise<void[]> {
    const eventHandler = this.fetchTransactionEventHandler(event.type);
    promises.push(
        ...eventHandler(event);
    )
  }
```

Reply:
Added in 9fb9f4ff1ae9ed9cd25837cc165356946b72e5b1.